File: tests/src/test/java/com/optivem/atddaccelerator/templategenerator/systemtest/util/process/ProcessResultAssertions.java
package com.optivem.atddaccelerator.templategenerator.systemtest.util.process;

import static org.assertj.core.api.Assertions.assertThat;

public class ProcessResultAssertions {

    public static void assertSuccess(ProcessResult result) {
        assertThat(result.isSuccess()).withFailMessage(() -> stringify(result)).isTrue();
    }

    public static void assertFailure(ProcessResult result) {
        assertThat(result.isError()).withFailMessage(() -> stringify(result)).isTrue();
    }

    public static void assertSuccess(ProcessResult result, String customMessage) {
        assertThat(result.isSuccess()).withFailMessage(() -> stringify(result, customMessage)).isTrue();
    }

    public static void assertFailure(ProcessResult result, String customMessage) {
        assertThat(result.isError()).withFailMessage(() -> stringify(result, customMessage)).isTrue();
    }

    private static String stringify(ProcessResult result) {
        return "Process finished with exit code " + result.getExitCode() + "\nErrors: " + result.getErrors() + "\nOutput: " + result.getOutput();
    }

    private static String stringify(ProcessResult result, String customMessage) {
        return customMessage + "\n" + stringify(result);
    }

}

File: tests/Util/Process/ProcessResultAssertions.cs
using Shouldly;

namespace Optivem.AtddAccelerator.TemplateGenerator.SystemTests.Util.Process
{
    public static class ProcessResultAssertions
    {
        public static void ShouldSucceed(this ProcessResult result)
        {
            result.IsSuccess.ShouldBeTrue(Stringify(result));
        }

        public static void ShouldSucceed(this ProcessResult result, string customMessage)
        {
            result.IsSuccess.ShouldBeTrue(Stringify(result, customMessage));
        }

        public static void ShouldFail(this ProcessResult result)
        {
            result.IsError.ShouldBeTrue(Stringify(result));
        }

        public static void ShouldFail(this ProcessResult result, string customMessage)
        {
            result.IsError.ShouldBeTrue(Stringify(result, customMessage));
        }

        private static string Stringify(ProcessResult result)
        {
            return $"Process finished with exit code {result.ExitCode}\nErrors: {result.Errors}\nOutput: {result.Output}";
        }

        private static string Stringify(ProcessResult result, string customMessage)
        {
            return $"{customMessage}\n{Stringify(result)}";
        }
    }
}